src/care/KeyValueSorter.h

```diff
@@ -223,6 +223,24 @@ class KeyValueSorter<T, RAJADeviceExec> {
          setFromArray(len, arr);
       }
 
+      ///////////////////////////////////////////////////////////////////////////
+      /// @author Alan Dayton
+      /// @brief Constructor
+      /// Allocates space and initializes the KeyValueSorter by copying
+      ///    elements and ordering from the given managed array
+      /// @param[in] len - The number of elements to allocate space for
+      /// @param[in] arr - The managed array to copy elements from
+      /// @return a KeyValueSorter instance
+      ///////////////////////////////////////////////////////////////////////////
+      KeyValueSorter<T, RAJADeviceExec>(const size_t len, const host_device_ptr<const T>& arr)
```

Comment:
You should simply be able to update the above constructor to accept `const host_device_ptr<const T>&` rather than adding another constructor.

Reply:
Done, changed setFromArray to have the same signature also